app/controllers/CashAccountV2Controller.scala

```diff
@@ -86,8 +87,13 @@ class CashAccountV2Controller @Inject()(authenticate: IdentifierAction,
       }
   }
 
-  def onSubmit(page: Option[Int]): Action[AnyContent] = Action.async {
-    Future.successful(NotImplemented("Needs to be implemented"))
+  def onSubmit(page: Option[Int]): Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
```

Comment:
We need to handle the form error scenarios as well. Right now we are directing to Page not found page and not a meaningful message or a page. Could you discuss this with Harsha and get a ticket raised for this?

Reply:
As discussed will be a part of another ticket, as there's no error validation now and too much work for this ticket to handle all the errors